src/ziggurat/util/java_util.clj

```diff
@@ -0,0 +1,59 @@
+(ns ziggurat.util.java-util
+  (:require [clojure.string :as str])
+  (:import (java.util Map Arrays)))
+
+(declare create-clojure-hash-map)
+(declare create-clojure-vector)
+(declare create-clojure-vector-from-array)
+
+(defn- is-java-array?
+  [array]
+  (str/starts-with? (.getName (.getClass array)) "["))
+
+(defn create-clojure-vector
+  [^java.lang.Iterable java-list]
+  (let [cloj-seq (seq java-list)]
+    (vec (map (fn [x]
```

Comment:
we can use [clojure.walk/postwalk](https://clojuredocs.org/clojure.walk/postwalk) instead of writing our own map

Reply:
+1